### ISSUE TITLE
Investigate and document buttons styled as links

### ISSUE THREAD

**Issue by trevorsaint** [ONSdigital/design-system#761]
I think we need a button that looks like a link. I also would expect `.btn--link` to be a modifier to do exactly that.

However we are using the modifier `.btn--link` to make links look like buttons. Do we really need to do this? Can’t we just assume that if an `<a>`, `<button>` has a class of `.btn` it will look like a button.

I do feel this is a much better approach, and we can use the modifier to have a `<button>` that looks like a link.

**GDS** do it this way and it does make more sense. 
https://design-system.service.gov.uk/components/button/

_Originally posted by @trevorsaint in https://github.com/ONSdigital/design-system/issues/45#issuecomment-636043593_

**Comment by boxadesign** [comment]
I have a couple of opinions on this...

Hi Trevor. 

In this instance it's the same as the GDS' `govuk-button--start`. I agree that it's kinda backwards as it's not a button it's a link so should really be `link--button`, we probably just copied the GDS convention + it's all held in the `onsButton` component which is something we could probably improve.

On the point of having buttons that look like links, this is something we've always stayed away from. Until very recently we didn't actually style links as buttons but felt in the scenario of a call to action it was justified. 

Reasons for not styling buttons as links are based on testing (particularly disabled users) and convention. We could probably dig out some accessibility testing from way back and obviously with `role=` we can get around some of the issues. 

I think the main reason is why would we want a button to look like a hyper link? Hyperlinks give the user a level of expectation and affordance that we haven't found a good reason why we would need/want a button to replicate that. Hyperlinks take you to a new page and we've been fairly dogmatic in keeping it to that.

**Comment by trevorsaint** [comment]
Hi Matt.

The start button modifier that GDS have is a modification on button style and not because it is a link.

They use the button class on `<buttons>` and `<link>` depending on its use case.

I get your point on having buttons that look like links.

It would be good to get your views on the following situation. I would expect 'Reset of filters' to be a button element and not a link element.

It may/may not cause issues. Perhaps the pattern needs to change?

**Filters concept for mobile (Census 2021 Engage phase Resources for Partners)**

![image](https://user-images.githubusercontent.com/4989027/83290119-09f3b680-a1de-11ea-97a4-011739f13084.png)

**Comment by boxadesign** [comment]
We use `btn--link` for the same intention as GDS (modify the style) but it's exclusively used for links. I think GDS only use start buttons as links also but I may be wrong. But if it's caused confusion it's worth changing the name.

With the screenshots above this could be a good case for introducing buttons as links.  Worth the wider design community looking at this.

**Comment by jrbarnes9** [comment]
I think it's a good case for styling an html <button> as a link.  

As long as we use aria `role=link` and use a hidden submit button at the top of the form to ensure the keyboard enter key submits the form rather than resetting it (we had to do this in EQ runner for 'clear selection' button for voluntary radio questions).

The main issue around links styled as buttons, and buttons styled as links, was around older versions of voice recognition software that did not support aria `role`.  But after discussions with DAC last summer we decided to allow 'launcher' links to be styled as buttons as long the expectation was clear to the user (the chevron icon on the button helped with this affordance).  

According to the last GDS assisted tech survey, the usage of these older versions was low.  DAC also suggested that the software upgrade process was more affordable (than buying new) and that frequent users would tend to upgrade due to the benefits of using the new version.

**Comment by armstrongb** [comment]
This will require extensive testing with voice control software as the button styled as link was a fail with Dragon when tested in eQ. If newer versions have resolved this with aria `role` support then this could be brought in as long as there was strong evidence to show that this was not contrary to user expectations and was not introducing an accessibility fragility.

Is there any evidence to show that the button styled reset filters performed any better or worse than the link styled one?

>Hyperlinks give the user a level of expectation and affordance that we haven't found a good reason why we would need/want a button to replicate that. Hyperlinks take you to a new page and we've been fairly dogmatic in keeping it to that.

**Comment by rmccar** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Issue #761: We need a button that looks like a link](https://trello.com/c/fYzcfuoV/697-issue-761-we-need-a-button-that-looks-like-a-link)
